programs/degen-pools/src/transfers.rs

```diff
@@ -0,0 +1,49 @@
+use crate::constants::AUTHORITY_PUBKEY;
+use crate::CustomError;
+use anchor_lang::prelude::*; 
+
+#[derive(Accounts)]
+pub struct ExecuteTransfer<'info> {
+    #[account(mut)]
+    pub sender: Signer<'info>,
+    /// CHECK: This is a safe account as it is the system
+    pub system_program: AccountInfo<'info>,
+    #[account(mut, address = AUTHORITY_PUBKEY)]
+    /// CHECK: The receiver's address is validated against a known authority.
+    pub receiver: AccountInfo<'info>, 
+}
+
+pub fn handle_transfer(ctx: Context<ExecuteTransfer>, amount: u64) -> Result<()> {
+    require!(
+        amount > 0,
+        CustomError::InvalidAmount
+    );
+
+    require!(
+        ctx.accounts.sender.lamports() >= amount,
+        CustomError::InsufficientFunds
```

Comment:
Is this check really needed? I assume the transfer would fail if the balance isn't enough. I'm just trying to reduce the bloat in the code so if we don't need the check, we shouldn't have it